src/huggingface_hub/inference/_client.py

```diff
@@ -143,12 +147,13 @@ class InferenceClient:
     def __init__(
         self,
         model: Optional[str] = None,
-        token: Union[str, bool, None] = None,
+        token: Optional[str] = None,
```

Comment:
If this is a voluntary change, you should update the docstring above as well

Reply:
(passing `False` would be unexpected

Reply:
Updated to revert to `Union[str, bool, None]` + fix related issue + updated docstrings in https://github.com/huggingface/huggingface_hub/pull/2094/commits/11c37a5a45e8b3b52e5126d5e7754a28b0759c39. It's more of a documentation / typing issue since bool value is always correctly respected (but in some cases I don't want to document it since it's not relevant when auth is required)